connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala

```diff
@@ -0,0 +1,472 @@
+    }
+
+    val finalCreateStmt = createWithReplication.withDurableWrites(ksMeta.getOrElse(DurableWrites, "True").toBoolean)
+
+    connector.withSessionDo { session =>
+      session.execute(finalCreateStmt.asCql())
+    }
+  }
+
+  override def alterNamespace(namespace: Array[String], changes: NamespaceChange*): Unit = {
+    checkNamespace(namespace)
+
+    val ksMeta: mutable.Map[String, String] = changes.foldLeft(loadNamespaceMetadata(namespace).asScala) {
+      case (metadata: mutable.Map[String, String], setProperty: SetProperty) =>
+        metadata + (setProperty.property() -> setProperty.value)
+      case (metadata: mutable.Map[String, String], removeProperty: RemoveProperty) =>
+        metadata - removeProperty.property()
+      case (_, other) => throw new CassandraCatalogException(s"Unable to handle alter namespace operation: ${other.getClass.getSimpleName}")
+    }
+
+    val alterStart = SchemaBuilder.alterKeyspace(namespace.head)
+    val alterWithDurable = alterStart.withDurableWrites(ksMeta.getOrElse(DurableWrites, "True").toBoolean)
+    val replicationClass = ksMeta
+      .getOrElse(ReplicationClass, throw new CassandraCatalogException(s"Altering a keyspace requires a $ReplicationClass option"))
+      .split("\\.")
+      .last
+    val alterWithReplication = replicationClass.toLowerCase(Locale.ROOT) match {
+      case SimpleStrategy =>
+        val replicationFactor = ksMeta.getOrElse(ReplicationFactor,
+          throw new CassandraCatalogException(s"Need a $ReplicationFactor option with SimpleStrategy"))
+        alterWithDurable.withSimpleStrategy(replicationFactor.toInt)
+      case NetworkTopologyStrategy =>
+        val datacenters = (ksMeta -- IgnoredReplicationOptions).map(pair => (pair._1, pair._2.toInt: java.lang.Integer))
+        alterWithDurable.withNetworkTopologyStrategy(datacenters.asJava)
+      case other => throw new CassandraCatalogException(s"Unknown replication strategy $other")
+    }
+
+    connector.withSessionDo(session =>
+      session.execute(alterWithReplication.asCql())
+    )
+  }
+
+  override def loadNamespaceMetadata(namespace: Array[String]): java.util.Map[String, String] = {
+    val ksMetadata = getKeyspaceMeta(connector, namespace)
+
+    (ksMetadata.getReplication.asScala + (DurableWrites -> ksMetadata.isDurableWrites.toString))
+      .asJava
+  }
+
+  override def dropNamespace(namespace: Array[String]): Boolean = {
+    checkNamespace(namespace)
+    val keyspace = getKeyspaceMeta(connector, namespace)
+    val dropResult = connector.withSessionDo(session => session.execute(SchemaBuilder.dropKeyspace(keyspace.getName).asCql()))
+    dropResult.wasApplied()
+  }
+
+  override def listTables(namespace: Array[String]): Array[Identifier] = {
+    getKeyspaceMeta(connector, namespace)
+      .getTables.asScala
+      .keys
+      .map { case tableName => Identifier.of(namespace, tableName.asInternal()) }
+      .toArray
+  }
```

Comment:
Semi-nit: I'm not sure the case statement is buying you much here... maybe this instead?

```scala
  override def listTables(namespace: Array[String]): Array[Identifier] = {
    getKeyspaceMeta(connector, namespace)
      .getTables.asScala
      .keys
      .map(DriverUtil.toName)
      .map(Identifier.of(namespace, _))
      .toArray
  }
```

See above re: extracting logic of a map of CqlIdentifier -> Any => Seq[String] from this function and listNamespaces() above.

Reply:
yeah I just popped out the case part, switched it to a single line map